replica/database.cc

```diff
@@ -1574,6 +1574,32 @@ lw_shared_ptr<memtable> memtable_list::new_memtable() {
     return make_lw_shared<memtable>(_current_schema(), *_dirty_memory_manager, _table_stats, this, _compaction_scheduling_group);
 }
 
+future<> memtable_list::clear_and_add() {
+    auto mt = new_memtable();
+    std::vector<replica::shared_memtable> new_memtables;
+    try {
+        // try preparing the cleared memtables vector ahead of time
+        // so we can clear the old memtables gently later
+        new_memtables.reserve(_memtables.size());
+        new_memtables.emplace_back(std::move(mt));
```

Comment:
Why do we need so many new memtables? Isn't one enough?

Reply:
Well, a reservation for so many new memtables.

Reply:
I didn't want to change the outcome of this function that kept the vector capacity (since it only called clear) in the synchronous path by calling `std::vector::clear()`.

Reply:
The question is we indeed want to keep that.
I agree that replacing the `_memtables` vector with a vector of size 1 seems reasonable and would be simpler.